### ISSUE TITLE
Test IAB CCPA Compliance Framework

### ISSUE THREAD

**Issue by SebastianZimmeck** [privacy-tech-lab/optmeowt-browser-extension#52]
As just discussed with @kalicki1, we want to put in a short paper entry at [WPES](https://wpes.tech/2020/). The idea is to make the claim that OptMeowt is able to automatically opt out users from sites that make use of the IAB CCPA Compliance Framework. In particular, we want to test for 20 sites whether it is possible to rewrite the `usprivacy` cookie. If it works for 20 sites, there is good reason to believe that it will work for any site that implements the IAB CCPA Framework.

Here is the test set, randomly selected from [cookiepedia](https://cookiepedia.co.uk/cookies/). I have not found a site with a `us-privacy` cookie.

1. https://www.islands.com/ (`usprivacy` cookie)
2. https://uk.ask.com/ (`us_privacy` cookie)
3. https://www.informationvine.com/ (`us_privacy` cookie)
4. https://www.evite.com/ (`us_privacy` cookie)
5. https://www.reference.com/ (`us_privacy` cookie)
6. https://www.dummies.com/ (`usprivacy` cookie)
7. https://www.reuters.com/ (`usprivacy` cookie)
8. https://www.thetradedesk.com/ (`usprivacy` cookie)
9. https://www.newyorkupstate.com/ (`usprivacy` cookie)
10. https://beautybox.allure.com/ (`usprivacy` cookie)
11. https://www.cnn.com/ (`usprivacy` cookie)
12. https://www.modbee.com/ (`usprivacy` cookie)
13. https://www.cruisingworld.com/ (`usprivacy` cookie)
14. https://www.edmunds.com/ (`usprivacy` cookie)
15. https://lamborghinichat.com/ (`usprivacy` cookie)
16. https://www.liveleak.com/ (`usprivacy` cookie)
17. https://www.streetmachinenationals.net/ (`usprivacy` cookie)
18. https://www.tartanroad.co.uk/ (`usprivacy` cookie)
19. https://imost.interestingthings.com/ (`usprivacy` cookie)
20. https://qa2.kansas.com/ (`usprivacy` cookie)

@kalicki1 , the way to approach this is to do preliminary tests with other sites not included in the 20. Then, if it works, for the other sites, run on the test set. If there are sites from the test set where it does not work, fix OptMeowt such that it works for those. Then, let me know, and I will pick up a new test set of 20 sites, and we run again until there are no more mistakes in the test set (we can probably tolerate one or two mistakes, but beyond that I would be skeptical).

Also, if the above test set contains a site that you already tested on in your preliminary testing, please pick a random site from cookiepedia and use that instead of the one that you already analyzed.

For every site in the test set the analysis procedure is as follows: 
1. The site is loaded with the `usprivacy` cookie as set.
2. Then, the cookie is modified via OptMeowt to opt out the user from selling. As I see it, the [usprivacy](https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/US%20Privacy%20String.md) cookie value should be `1YYN`. (Ideally we would be able to read the cookie value and retain some of its value, e.g., in the fourth position; whatever is written there remains; however, this is something we do not need to address immediately; for the time being, writing `1YYN` is enough.)
3. The site is reloaded and the cookie still has the value `1YYN`, i.e., the user is opted out.

As we discussed, @kalicki1, let's not worry about domains vs subdomains. For example, if a test set site is a subdomain, we only require the change there (it can be on the domain as well, but that is not required for our test). 

(cc'ing @rgoldstein01 and @davebaraka)

**Comment by kalicki** [comment]
Hey guys! for some reason I`m receiving notifications from you, perhaps because it is a similar username with @kalicki1?! How funny!

**Comment by SebastianZimmeck** [comment]
Thanks for checking in @kalicki! Typo on my end. Sorry about that ... :)

**Comment by SebastianZimmeck** [comment]
@kalicki1 provided a [Google Sheet](https://docs.google.com/spreadsheets/d/1JV0ai_oN0EFjKdzOPVWlGlVA0rbF5QlLiVl0m75Zy3M/edit?usp=sharing) to keep track of the tests.

**Comment by kalicki1** [comment]
**Regarding the above commit and tests so far:**

When I finished testing the first 20 links @SebastianZimmeck provided above, I found two instances of a site's IAB cookie not being read properly, and hence not updated properly. I managed to find that this was related to the fact that both of these sites had their IAB cookies set to a non-default URL path (i.e. not "/"). Changing OptMeowt's IAB cookie handler to use the site's full URL instead of the URL origin solved this issue. 

I also updated the IAB signal validator to account for signals with combinations of hyphens and letters, since one site seemed to be using this form of signal (i.e. `1Y-N`). This could be something I can come back to in the future to polish up. 

This is the last change that was necessary following this initial round of tests, so now I can move forward with a new round of tests.
